src/fidesops/models/saasconfig.py

```diff
@@ -0,0 +1,142 @@
+    """
+
+    name: str
+    type: Literal[
+        "query", "path", "body"
+    ]  # used to determine location in the generated request
+    default_value: Optional[Any]
+    identity: Optional[str]
+    data_type: Optional[str]
+    references: Optional[List[FidesopsDatasetReference]]
+
+
+class Strategy(BaseModel):
+    """General shape for swappable strategies (ex: auth, pagination, postprocessing, etc.)"""
+
+    strategy_name: str
+    configuration: Dict[str, Any]
+
+
+class Request(BaseModel):
+    """
+    A single request with a static or dynamic path, and the request params needed to build the request.
+    Also includes optional startegies for pre/postprocessing and pagination
+    """
+
+    path: str
+    request_params: Optional[List[RequestParam]]
+    data_path: Optional[str]  # defaults to collection name if not specified
+    preprocessor: Optional[Strategy]
+    postprocessor: Optional[Strategy]
+    pagination: Optional[Strategy]
+
+
+class Endpoint(BaseModel):
+    """An collection of read/update/delete requests which corresponds to a FidesopsDataset collection (by name)"""
+
+    name: str
+    requests: Dict[Literal["read", "update", "delete"], Request]
+
+
+class ConnectorParam(BaseModel):
+    """Used to define the required parameters for the connector (user-provided and constants)"""
+
+    name: str
+    default_value: Optional[Any]
+    from_user: bool
+    is_secret: bool
+
+
+class ConnectorParamRef(BaseModel):
+    """A reference to a value in the connector params (by name)"""
+
+    connector_param: str
+
+
+class ClientConfig(BaseModel):
+    """Definition for an authenticated base HTTP client"""
+
+    protocol: str
+    host: Union[
+        str, ConnectorParamRef
+    ]  # can be defined inline or be a connector_param reference
+    authentication: Strategy
+
+
+class SaaSConfig(BaseModel):
+    """
+    Used to store endpoint and param configurations for a SaaS connector.
+    This is done to separate the details of how to make the API calls
+    from the data provided by a given API collection.
+
+    The required fields for the config are converted into a Dataset which is
+    merged with the standard Fidesops Dataset to provide a complete set of dependencies
+    for the graph traversal
+    """
+
+    fides_key: FidesOpsKey
+    name: str
+    description: str
+    version: str
+    connector_params: List[ConnectorParam]
+    client_config: ClientConfig
+    endpoints: List[Endpoint]
+    test_request: Request
+
+    @property
+    def top_level_endpoint_dict(self) -> Dict[str, Endpoint]:
+        """Returns a map of endpoint names mapped to Endpoints"""
+        return {endpoint.name: endpoint for endpoint in self.endpoints}
+
+    def generate_dataset(self) -> Dataset:
+        """Converts endpoints to a Dataset with collections and field references"""
+        collections = []
+        for endpoint in self.endpoints:
+            fields = []
```

Comment:
This feels a bit murky, in your dataset then, endpoints become collections and request_params become fields.   Would it be worth calling them that in your SaasConfig, or noting the parallels here? The fact that these endpoints become nodes is not really transparent based on the config. Maybe just better documentation around the SaasConfig here.

Reply:
I'll include this in the documentation. I want to keep the current endpoint/request terminology in the SaaS config if possible.